packages/gatsby/src/utils/page-data.js

```diff
@@ -0,0 +1,21 @@
+const fs = require(`fs-extra`)
+const path = require(`path`)
+
+const getFilePath = ({ publicDir }, pagePath) => {
+  const fixedPagePath = pagePath === `/` ? `index` : pagePath
+  return path.join(publicDir, `page-data`, fixedPagePath, `page-data.json`)
+}
+
+const write = async ({ publicDir }, page, result) => {
+  const filePath = getFilePath({ publicDir }, page.path)
+  const body = {
+    componentChunkName: page.componentChunkName,
+    path: page.path,
+    result,
+  }
+  await fs.outputFile(filePath, JSON.stringify(body))
```

Comment:
we should move to V8 serialize here too
https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/redux/persist.js
```js
const useV8 = Boolean(v8.serialize)
const [serialize, deserialize, file] = useV8
  ? [v8.serialize, v8.deserialize, `${process.cwd()}/.cache/redux.state`]
  : [jsonStringify, jsonParse, `${process.cwd()}/.cache/redux-state.json`]
```

It might be good to create a simple serialize utility that does the v8.serialize check.

Feel free to merge, as this is not blocking. Could be a follow up

Reply:
Just catching up on v8.serialize. It looks like it serializes using a non-JSON binary format. But the `page-data.json` needs to be loaded in the browser, so I don't think it will work for this use case.

Reply:
oh yeah, that's true.

Reply:
huzzah!